### PR TITLE
fix: Fix could not fetch Ollama model

### PR DESCRIPTION
**Pull request by leehuwuj**

### CODE REVIEW

File: src/controllers/providers.py [hunk 1]
```diff
@@ -4,20 +4,22 @@
...
     @staticmethod
-    def fetch_ollama_models() -> List[str]:
+    def fetch_ollama_models(provider_url: str = None) -> List[str]:
         """
         Fetch all available models from the Ollama provider.
         """
         from ollama import Client
 
-        client = Client(host=os.getenv("OLLAMA_BASE_URL", "http://127.0.0.1:11434"))
+        client = Client(host=provider_url or os.getenv("OLLAMA_URL"))
```
Comment (marcusschiesser) [inline src/controllers/providers.py:13]: why `OLLAMA_URL`?
```suggestion
        client = Client(host=provider_url or os.getenv("OLLAMA_BASE_URL"))
```

File: admin/client/config.ts [hunk 2]
```diff
@@ -155,9 +155,12 @@ export async function updateConfig(
...
 }
 
-export async function fetchModels(provider: string): Promise<string[]> {
+export async function fetchModels(
+  provider: string,
+  providerUrl: string,
+): Promise<string[]> {
   const res = await fetch(
-    `${getBaseURL()}/api/management/config/models?provider=${provider}`,
+    `${getBaseURL()}/api/management/config/models?provider=${provider}&provider_url=${providerUrl}`,
```
Comment (marcusschiesser) [inline admin/client/config.ts:163]: don't you need `encodeURIComponent` to encode the GET parameter?
```suggestion
    `${getBaseURL()}/api/management/config/models?provider=${provider}&provider_url=${encodeURIComponent(providerUrl)}`,
```

File: src/controllers/providers.py (hunk 1 above)
Comment (leehuwuj) [reply src/controllers/providers.py:13]: ah, sorry it should be OLLAMA_BASE_URL

File: admin/client/config.ts (hunk 2 above)
Comment (leehuwuj) [reply admin/client/config.ts:163]: just a base url so i think it'll alright. If you change to encodeURIComponent then you also need to decode it in python

**Review by marcusschiesser — APPROVED** [review]

**Pull request merged**